coding_guidelines_c++.md

```diff
@@ -12,27 +12,8 @@ In the code review process we want to enforce a consistent coding style to impro
 
```

Comment:
This file had `_c++` in its name and links to it point to C++ code guidelines, but the header just says "NEST Coding Style Guidelines". I think the header should mention C++, and we should have links to our other guides (SLI, Python). We might also add a section on Python here.